etc/calico/confd/templates/bird6.cfg.template

```diff
@@ -194,6 +197,9 @@ protocol bgp Global_{{$id}} from bgp_template {
 {{if ls $node_peers_key}}
 {{range gets (printf "%s/*" $node_peers_key)}}{{$data := json .Value}}
 {{$nums := split $data.ip ":"}}{{$id := join $nums "_"}}
+{{- if $data.port}}
+{{- $id = printf "%s_%.0f" $id $data.port}}
```

Comment:
Could this algorithm give collisions between IDs for IPv6 peers with no port and ones with a port?  I'm worried that the colon of the IPv6 address gets converted to `_` too so maybe there's a way to make them collide.  Quick fix would be to do ```
$id = printf "%s_port_%.0f" $id $data.port
```
so it's totally unambiguous.

Reply:
yes, that will be a good way to differentiate between port and IP, I'll update it for both IPv6 and IPv4